### ISSUE TITLE
Gifting to a player skips one turn before it arrives

### ISSUE THREAD

**Issue by zenseii** [ihhub/fheroes2#5252]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

Playing hotseat mode and controlling all players, if you gift resource then they will not arrive on the next turn of the receiving players but on their second turn after the gifting is done.

### Describe the possible solution

The gift should arrive on the first turn of the receiving player after the gift is sent.

**Comment by oleg-derevenetz** [comment]
Hi @zenseii gifted resources will arrive on the next DAY. Please don't mix DAY and TURN :) While the red player has a TURN after the blue player, it is still the same DAY, so he will need to wait for the next DAY for gifted resources to arrive.

**Comment by zenseii** [comment]
Right, thanks! I corrected it.

**Comment by oleg-derevenetz** [comment]
@zenseii I understand the logic of those who developed the gift mechanics - if you make it so that resources come just on the next turn, then sometimes they will come on the same day, and sometimes on the next day (depending on the sequence of the giving player's and the receiving player's turn), because time cannot go back. Now everything happens simply and logically - moving resources from one player to another always takes one day, and receiving player will always get them in his turn on the next day. I propose to leave this logic as it is.

**Comment by Branikolog** [comment]
Hello, @zenseii .
I agree with @oleg-derevenetz . We should keep clear and obvious logic for this "gift" mechanics.
While playing with allies there could be advantage for the players, that start earlier, which is unfair.

**Comment by zenseii** [comment]
> Hello, @zenseii . I agree with @oleg-derevenetz . We should keep clear and obvious logic for this "gift" mechanics. While playing with allies there could be advantage for the players, that start earlier, which is unfair.

Yes, this is true. I already thought about there being an advantage if you are first in the turn order. However, this is also the case for many other things regarding the game and the turn order. There are also disadvantages to being the first player.

Personally, I don't mind it being either way. Balancing aspects like these can only be done after lots of playtesting, and for this aspect such a thing will only be possible after online multiplayer has been implemented, because the amount of players playing in hotseat and against each other would be quite few.

**Comment by ihhub** [comment]
Shall we just state that resource gifting takes 1 day? Something like "Resource Guild needs 1 day to process your request".

**Comment by zenseii** [comment]
> Shall we just state that resource gifting takes 1 day? Something like "Resource Guild needs 1 day to process your request".

Sure. Where would we put this?

**Comment by ihhub** [comment]
I think in the same window where we select recipients. We might need to expand the window.
